File: SoloLearn Codes/Longest Common Substring/LongestCommonSubstring.cs
using System;
using System.Linq;

namespace SoloLearn
{
    class Program
    {
        static void Main(string[] args)
        {
            string[] ws = Console.ReadLine().Split();
            string fw = ws[0];
            string lcs = $"{fw[0]}";

            for (int i = 0; i < fw.Length; i++)
            {
                for (int j = i + 2; j < fw.Length + 1; j++)
                {
                    string sub = fw.Substring(i, j - i);
                    if (ws.Where(x => x.Contains(sub)).Count() == ws.Length)
                    {
                        if (sub.Length > lcs.Length) lcs = sub;
                    }
                }
            }

            Console.WriteLine(lcs);
        }
    }
}

File: SoloLearn Codes/Longest Common Substring/longest_common_substring.swift
import Foundation

var ws = readLine()!.split(separator: " ")
var fw = ws[0]
var lcs = String(Array(fw)[0])

for i in 0..<fw.count {
    for j in (i + 2)..<(fw.count + 1) {
        let sub = String(Array(fw)[i..<j])
        if ws.filter({ $0.range(of: sub) != nil }).count == ws.count {
            if sub.count > lcs.count {
                lcs = sub
            }
        }
    }
}

print(lcs)